File: easy/22. Caesar Cipher Encryption/caesar_cipher_encryption.go
package main

func CaesarCipherEncryptor(str string, key int) string {
	result := ""
	for _, char := range str {
		if char >= 'a' && char <= 'z' {
			result += string(CaesarCipherEncryptorChar(char, key))
		} else if char >= 'A' && char <= 'Z' {
			result += string(CaesarCipherEncryptorChar(char, key))
		} else {
			result += string(char)
		}
	}

	return result
}

func CaesarCipherEncryptorChar(char rune, key int) rune {
	return rune((int(char)+key-97)%26 + 97)
}

File: dsa/easy/Caesar Cipher Encryption/caesar_cipher_encryption.cpp
#include <string>

std::string caesarCypherEncryptor(std::string str, int key) {
  std::string result = "";
  for (int i = 0; i < str.length(); i++) {
    if (str[i] >= 'a' && str[i] <= 'z') {
      result += (str[i] + key - 'a') % 26 + 'a';
    } else if (str[i] >= 'A' && str[i] <= 'Z') {
      result += (str[i] + key - 'A') % 26 + 'A';
    } else {
      result += str[i];
    }
  }
  return result;
}

